### PR TITLE
First release of Oracle GoldenGate Veridata docker scripts

### PR DESCRIPTION
**Pull request by arnabnandioracle**
Initial commit

Signed-off-by: Arnab Nandi <arnab.x.nandi@oracle.com>

### CODE REVIEW

File: OracleVeridata/12.2.1.4/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+3) OGG Veridata Installer
+
+Download the OGG Veridata 12.2.1.4.0 Installer.
+
+https://www.oracle.com/middleware/technologies/goldengate-downloads.html
+
+4) Latest Bundle Patch (optional but recommended)
+ 
+Download the latest bundle patch.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:31]: From where? How? Where should I put it?

File: OracleVeridata/12.2.1.4/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+5) Run buildDockerImage.sh
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:33]: Needs to be changed to `buildContainerImage.sh` and formatted as a heading

File: OracleVeridata/12.2.1.4/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+Usage: buildDockerImage.sh -v [version]
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:35]: Change to `buildContainerImage.sh` and the entire output needs to be formatted as a code sample.

File: OracleVeridata/12.2.1.4/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+Builds a Docker Image for Oracle GoldenGate Veridata.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:36]: Change to `Builds a container image for Oracle GoldenGate Veridata.`

File: OracleVeridata/12.2.1.4/README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+-i: OGG Veridata Installer zip file
+-f: FMW Release version.Default is 12.2.1.4-210701
+-p: Patch file
+-h: Help
+
+e.g.
+buildDockerImage.sh -v 12.2.1.4-210630 -i fmw_12.2.1.4.0_ogg_Disk1_1of1.zip -p p32761281_122140_Generic.zip
+
+
+How to run OGG Veridata Server
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:49]: Needs to be formatted as a heading

File: OracleVeridata/12.2.1.4/README.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+1) Create Docker Bridge
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:51]: Needs to be formatted as heading. Also, you're not creating a bridge, you're creating a _network_ of type `bridge` so this should be "Create Docker network".

Also, you should explain why this network is necessary. What's the purpose of it?

File: OracleVeridata/12.2.1.4/README.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+e.g.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:53]: Remove this line.

File: OracleVeridata/12.2.1.4/README.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+docker network create -d bridge VdtBridge
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:54]: Format as a code sample

File: OracleVeridata/12.2.1.4/README.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+2) Oracle Database
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:56]: Format as a heading

File: OracleVeridata/12.2.1.4/README.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+Make sure the Database is running on the network ,created earlier, using --network.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:58]: Make sure what database is running on which network? You provided example commands before, why are they missing for this step? How does the user connect to an external Database that's not running as a container or even on this host?

File: OracleVeridata/12.2.1.4/README.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+3) OGG Veridata Admin Server
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:60]: Format this as a heading

File: OracleVeridata/12.2.1.4/README.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+Edit vdt.env file and execute run_admin_server.sh.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:62]: The filenames should be wrapped in back ticks as they are literals. Also, you must explain what needs to be edited and why and where to run the script. And what that script does. And how to run the image without a script.

File: OracleVeridata/12.2.1.4/README.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+4) OGG Veridata Managed Server
+
+Edit vdt.env file and execute run_managed_server.sh.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:66]: Are these the same edits as the previous step? Do I change my edits? How many copies of this file exist?

File: OracleVeridata/12.2.1.4/README.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+2) Oracle Database
+
+Make sure the Database is running on the network ,created earlier, using --network.
+
+3) OGG Veridata Admin Server
+
+Edit vdt.env file and execute run_admin_server.sh.
+
+4) OGG Veridata Managed Server
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:64]: Format as a heading

File: OracleVeridata/12.2.1.4/README.md [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+Edit vdt.env file and execute run_managed_server.sh.
+
+
+5) OGG Veridata Agent
+
+Edit vdtagent.env and execute run_agent.sh.
+
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:72]: Remove all the blank lines

File: OracleVeridata/12.2.1.4/README.md [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+3) OGG Veridata Admin Server
+
+Edit vdt.env file and execute run_admin_server.sh.
+
+4) OGG Veridata Managed Server
+
+Edit vdt.env file and execute run_managed_server.sh.
+
+
+5) OGG Veridata Agent
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:69]: Format as a heading

File: OracleVeridata/12.2.1.4/README.md [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+
+Edit vdtagent.env and execute run_agent.sh.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/README.md:71]: Same as previous. Formatting and context are missing.

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 18]
```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+#
+# Since: February, 2020
+# Description: script to build a Docker image for Oracle GoldenGate Veridata.
+#
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+#
+# Copyright (c) 2020-2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:9]: Remove "All rights reserved" as Oracle no longer uses this phrase.

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+#
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:11]: Change the URL to use `https://`

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 20]
```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:1]: This file needs to be renamed to `buildContainerImage.sh`

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+#
+# Copyright (c) 2020-2021 Oracle and/or its affiliates. All rights reserved.
+#
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+#
+#
+usage() {
+cat << EOF
+
+Usage: buildDockerImage.sh -v [version]
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:17]: change to `buildContainerImage.sh`

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 22]
```diff
@@ -0,0 +1,167 @@
...
+Builds a Docker Image for Oracle GoldenGate Veridata.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:18]: Change `Docker Image` to `container image`

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 23]
```diff
@@ -0,0 +1,167 @@
...
+
+Parameters:
+   -v: Release version to build. Default is 12.2.1.4.0
+   -i: OGG Veridata Installer zip file
+   -f: FMW Release version . Default is 12.2.1.4-210701
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:23]: remove the space before the `.`

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 24]
```diff
@@ -0,0 +1,167 @@
...
+   -p: Patch file
+   -h: Help
+
+
+
+Copyright (c) 2020-2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:29]: Remove "All rights reserved"

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 25]
```diff
@@ -0,0 +1,167 @@
...
+Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:30]: Change the URL to `https://`

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 26]
```diff
@@ -0,0 +1,167 @@
...
+  exit 1
+}
+BUILD_END=$(date '+%s')
+BUILD_ELAPSED=`expr $BUILD_END - $BUILD_START`
+
+echo ""
+
+if [ $? -eq 0 ]; then
+cat << EOF
+  Oracle GoldenGate Veridata Docker Image for version: $VERSION.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:158]: Change `Docker Image` to `container image`

File: OracleVeridata/12.2.1.4/buildDockerImage.sh [hunk 27]
```diff
@@ -0,0 +1,167 @@
...
+
+    --> $IMAGE_NAME
+
+  Build completed in $BUILD_ELAPSED seconds.
+
+EOF
+else
+  echo "Oracle GoldenGate Veridata Docker Image was NOT successfully created. Check the output and correct any reported problems with the docker build operation."
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/buildDockerImage.sh:166]: Change `Docker Image` to `container image` and remove the word `docker`.

File: OracleVeridata/12.2.1.4/container-scripts/applyRollbackPatch.sh [hunk 28]
```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+#
+#
+# Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/applyRollbackPatch.sh:4]: Remove the phrase "All rights reserved"

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtAgent.sh [hunk 29]
```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+#
+#
+# Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtAgent.sh:4]: Remove "All rights reserved"

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 30]
```diff
@@ -0,0 +1,281 @@
+#!/bin/bash
+#
+#
+# Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:4]: Remove "All rights reserved"

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 31]
```diff
@@ -0,0 +1,281 @@
...
+echo "Start the Admin and Managed Servers  "
+echo "====================================="
+
+# Check that the User has passed on all the details needed to configure this image
+# Settings to call RCU....
+
+
+echo "SCHEMA_PREFIX=${SCHEMA_PREFIX:?"Please set SCHEMA_PREFIX for the database schemas"}"
+echo "VERIDATA_DOMAIN_NAME=${VERIDATA_DOMAIN_NAME:?"Please set VERIDATA_DOMAIN_NAME for creating the new Domain"}"
+#echo "VERIDATA_PASSWORD=${VERIDATA_PASSWORD:?"Please set VERIDATA_PASSWORD"}"
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:59]: Remove this line

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 32]
```diff
@@ -0,0 +1,281 @@
...
+echo "VERIDATA_USER=${VERIDATA_USER:?"Please set VERIDATA_USER"}"
+echo "DATABASE_HOST=${DATABASE_HOST:?"Please set DATABASE_HOST"}"
+echo "DATABASE_PORT=${DATABASE_PORT:?"Please set DATABASE_PORT"}"
+echo "DATABASE_USER=${DATABASE_USER:?"Please set DATABASE_USER"}"
+#echo "DATABASE_PASSWORD=${DATABASE_PASSWORD:?"Please set DATABASE_PASSWORD"}"
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:64]: Remove this line

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 33]
```diff
@@ -0,0 +1,281 @@
...
+    VERIDATA_PASSWORD=${f_str}${s_str}
+    echo ""
+    echo "    Weblogic Administrator password Auto Generated :"
+    echo ""
+    echo "    ----> Weblogic Administrator password: $VERIDATA_PASSWORD"
+    echo ""
+fi
+
+
+#echo "DATABASE_JDBC_URL=${DATABASE_JDBC_URL} "
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:100]: Remove this line

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 34]
```diff
@@ -0,0 +1,281 @@
...
+
+if [ -z "${DATABASE_JDBC_URL}" ]
+then
+    CONNECTION_STRING="${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_SERVICE}"
+    #JDBC_URL="jdbc:oracle:thin:@$CONNECTION_STRING"
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:105]: Remove this line

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 35]
```diff
@@ -0,0 +1,281 @@
...
+  JDBC_URL="jdbc:weblogic:sqlserver://$CONNECTION_STRING"
+else
+  if [ "${DATABASE_TYPE}" == "MYSQL" ]
+  then
+    JDBC_URL="jdbc:mysql://$CONNECTION_STRING"
+  fi
+fi
+
+
+#CONNECTION_STRING="${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_SERVICE}"
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:128]: Remove this line and the next one

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 36]
```diff
@@ -0,0 +1,281 @@
...
+# set environments needed for the script to work
+
+CONFIGURE_DOMAIN="true"
+
+if [ -f ${DOMAIN_HOME}/servers/${ADMIN_NAME}/logs/${ADMIN_NAME}.log ];
+then
+    CONFIGURE_DOMAIN="false"
+fi
+
+#CONFIGURE_DOMAIN="true"
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:226]: Remove this line and the next one

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 37]
```diff
@@ -0,0 +1,281 @@
...
+	   	touch $CONTAINERCONFIG_DIR/VDT.Domain.Configure.suc
+      echo "$ORACLE_HOME/oracle_common/common/bin/wlst.sh -skipWLSModuleScanning $ORACLE_HOME/container-scripts/createVeridataDomain.py -oh $ORACLE_HOME -jh $JAVA_HOME -dh $DOMAIN_HOME -name $VERIDATA_DOMAIN_NAME -port $VERIDATA_PORT -user $VERIDATA_USER -password $VERIDATA_PASSWORD -rcuDb $JDBC_URL -rcuPrefix $SCHEMA_PREFIX -rcuSchemaPwd $SCHEMA_PASSWORD -adminPort $ADMIN_PORT -adminName $ADMIN_NAME -prodMode $PROD_MODE">> $CONTAINERCONFIG_DIR/VDT.Domain.Configure.suc
+	    echo "DATABASE_JDBC_URL=$JDBC_URL" > $CONTAINERCONFIG_DIR/vdtserverenv.sh
+	    echo "CONNECTION_STRING=$CONNECTION_STRING" > $CONTAINERCONFIG_DIR/vdtserverenv.sh
+	   	echo "SCHEMA_PREFIX=$SCHEMA_PREFIX" >> $CONTAINERCONFIG_DIR/vdtserverenv.sh
+	   	echo "SCHEMA_PASSWORD=$SCHEMA_PASSWORD" >> $CONTAINERCONFIG_DIR/vdtserverenv.sh
+
+	   	# Create the security file to start the server(s) without the password prompt
+      mkdir -p ${DOMAIN_HOME}/servers/${ADMIN_NAME}/security/
+      echo "username=${VERIDATA_USER}" >> ${DOMAIN_HOME}/servers/${ADMIN_NAME}/security/boot.properties
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:254]: This is not an acceptable way to store user credentials.

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 38]
```diff
@@ -0,0 +1,281 @@
...
+      echo "password=${VERIDATA_PASSWORD}" >> ${DOMAIN_HOME}/servers/${ADMIN_NAME}/security/boot.properties
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:255]: This is not an acceptable method of storing user credentials

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 39]
```diff
@@ -0,0 +1,281 @@
...
+	$ORACLE_HOME/oracle_common/common/bin/wlst.sh -skipWLSModuleScanning $ORACLE_HOME/container-scripts/createVeridataDomain.py -oh $ORACLE_HOME -jh $JAVA_HOME -dh $DOMAIN_HOME -name $VERIDATA_DOMAIN_NAME -port $VERIDATA_PORT -user $VERIDATA_USER -password $VERIDATA_PASSWORD -rcuDb $JDBC_URL -rcuPrefix $SCHEMA_PREFIX -rcuSchemaPwd $SCHEMA_PASSWORD -adminPort $ADMIN_PORT -adminName $ADMIN_NAME -prodMode $PROD_MODE
+	retval=$?
+	if [ $retval -ne 0 ]
+	then
+	   	echo "Domain Configuration failed.. please check the logs for errors"
+	   	exit
+	else
+	   	# Write the Domain suc file...
+	   	touch $CONTAINERCONFIG_DIR/VDT.Domain.Configure.suc
+      echo "$ORACLE_HOME/oracle_common/common/bin/wlst.sh -skipWLSModuleScanning $ORACLE_HOME/container-scripts/createVeridataDomain.py -oh $ORACLE_HOME -jh $JAVA_HOME -dh $DOMAIN_HOME -name $VERIDATA_DOMAIN_NAME -port $VERIDATA_PORT -user $VERIDATA_USER -password $VERIDATA_PASSWORD -rcuDb $JDBC_URL -rcuPrefix $SCHEMA_PREFIX -rcuSchemaPwd $SCHEMA_PASSWORD -adminPort $ADMIN_PORT -adminName $ADMIN_NAME -prodMode $PROD_MODE">> $CONTAINERCONFIG_DIR/VDT.Domain.Configure.suc
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:246]: Why are you `echo`'ing this?

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 40]
```diff
@@ -0,0 +1,281 @@
...
+      # Setting env variables
+      #=======================
+      echo ".${DOMAIN_HOME}/bin/setDomainEnv.sh" >> ${DOMAIN_HOME}/.bashrc
+      echo "export PATH=$PATH:${ORACLE_HOME}/common/bin:${DOMAIN_HOME}/bin" >> $ORACLE_HOME/.bashrc
+	fi
+fi
+
+#Set Java options
+export JAVA_OPTIONS=${JAVA_OPTIONS}
+echo "Java Options: ${JAVA_OPTIONS}"
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:267]: Why are you `echo`'ing this?

File: OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh [hunk 41]
```diff
@@ -0,0 +1,281 @@
...
+
+echo "Starting the Admin Server"
+echo "=========================="
+
+${DOMAIN_HOME}/bin/setDomainEnv.sh
+
+# Start Admin Server and tail the logs
+${DOMAIN_HOME}/startWebLogic.sh
+#touch ${DOMAIN_HOME}/servers/${ADMIN_NAME}/logs/${ADMIN_NAME}.log
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createOrStartVdtDomain.sh:276]: Remove this line.

File: OracleVeridata/12.2.1.4/container-scripts/createVeridataDomain.py [hunk 42]
```diff
@@ -0,0 +1,210 @@
+#
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createVeridataDomain.py:1]: You're missing a shebang line. All scripts must have the interpreter defined on the first line.

File: OracleVeridata/12.2.1.4/container-scripts/createVeridataDomain.py [hunk 43]
```diff
@@ -0,0 +1,210 @@
...
+# Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/container-scripts/createVeridataDomain.py:2]: Remove "All rights reserved"

**Review by Djelibeybi — CHANGES_REQUESTED** [review]
There are lots of changes necessary to meet the requirements for this repo.  I stopped reviewing after 70 comments. Please review the rest of the files I skipped and make the same fixes to those.

**Comment by Djelibeybi** [comment]
You need to remove all commented out lines, all debugging lines, most references to Docker (except when used as a specific example) and make sure all your Markdown files are formatted correctly. This is considered Oracle documentation so it should be at least reviewed by your doc team for adherence to all doc standards.

File: OracleVeridata/12.2.1.4/LICENSE.txt [hunk 44]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply OracleVeridata/12.2.1.4/LICENSE.txt:1]: Please remove this file from your pull request.

File: OracleVeridata/12.2.1.4/run_admin_server.sh [hunk 45]
```diff
@@ -0,0 +1,73 @@
...
+   fi
+}
+
+set_context
+
+extract_env WEBLOGIC_PORT ${scriptDir}/vdt.env
+extract_env DOMAIN_HOST_VOLUME ${scriptDir}/vdt.env
+extract_env ADMIN_CONTAINER_NAME ${scriptDir}/vdt.env
+extract_env ADMIN_CONTAINER_NAME ${scriptDir}/vdt.env
+extract_env ADMIN_CONTAINER_NAME ${scriptDir}/vdt.env
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/run_admin_server.sh:35]: Do you need to run this three times?

File: OracleVeridata/12.2.1.4/run_admin_server.sh [hunk 46]
```diff
@@ -0,0 +1,73 @@
...
+#extract_env DOMAIN_HOST_VOLUME ${scriptDir}/vdt.env
```
Comment (Djelibeybi) [inline OracleVeridata/12.2.1.4/run_admin_server.sh:36]: If you don't need this line, remove it.

File: OracleVeridata/12.2.1.4/README.md [hunk 47]
```diff
@@ -70,13 +59,13 @@ Parameters:
 For example: 
 `buildDockerImage.sh -v 12.2.1.4-210630 -i fmw_12.2.1.4.0_ogg_Disk1_1of1.zip -p p32761281_122140_Generic.zip`
```
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply OracleVeridata/12.2.1.4/README.md:60]: You need to change the example to `buildContainerImage.sh`

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

File: OracleVeridata/12.2.1.4/README.md [hunk 48]
```diff
@@ -0,0 +1,81 @@
...
+
+Oracle Database is required to install OGG Veridata repository.You can use an existing Oracle Database or pull Oracle DB Image.
+
+docker pull container-registry.oracle.com/database/enterprise:19.3.0.0
+
+3) OGG Veridata Installer
+
+Download the OGG Veridata 12.2.1.4.0 Installer.
+
+https://www.oracle.com/middleware/technologies/goldengate-downloads.html
```
[the remark this answers is not in the document]
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/README.md:37]: The instruction is verified from our Documentation Team.

File: OracleVeridata/12.2.1.4/buildContainerImage.sh [hunk 49]
```diff
@@ -0,0 +1,167 @@
...
+# BUILDING THE IMAGE #
+# ################## #
+echo "Building image '$IMAGE_NAME' ..."
+echo "Proxy Settings '$PROXY_SETTINGS'"
+# BUILD THE IMAGE (replace all environment variables)
+
+
+BUILD_START=$(date '+%s')
+
+docker build --build-arg VERIDATA_VERSION=${VERIDATA_VERSION} --build-arg INSTALLER_VERSION=${INSTALLER_VERSION} --build-arg INSTALLER=${VDT_INSTALLER} --build-arg FMW_VERSION=${FMW_VERSION} --build-arg PATCH_FILE=${PATCH_FILE} --build-arg OWNER_GROUP=${owner_group} --force-rm=true --no-cache=true $PROXY_SETTINGS -t $IMAGE_NAME -f Dockerfile . || {
```
[the remark this answers is not in the document]
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/buildContainerImage.sh:147]: Adhering to the existing release of Oracle FMW Infrastructure. Please follow the section "Build the FMW Infrastructure Image" in the link

https://github.com/oracle/docker-images/tree/main/OracleFMWInfrastructure/dockerfiles/12.2.1.4

File: OracleVeridata/12.2.1.4/README.md [hunk 50]
```diff
@@ -1,51 +1,58 @@
...
-## Prerequisites
+This readme accompanies the Oracle GoldenGate Veridata Docker image.
 
-Create Docker network to run Admin,Managed server and repository Database (in case needed)
+=============================================================================
 
-`docker network create -d bridge VdtBridge`
+## 1†Overview
+
+This Docker configuration has been used to create the Oracle GoldenGate Veridata image. The README facilitates the configuration and environment set up for DevOps users. This project includes the creation of an Oracle GoldenGate Veridata domain and agent.
```
[the remark this answers is not in the document]
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/README.md:6]: Read the heading section of "Oracle FMW Infrastructure" spinet pasted here :

"
This Docker configuration has been used to create the Oracle Fusion Middleware Infrastructure image. Providing this FMW image facilitates the configuration and environment set up for DevOps users. This project includes the creation of an FMW Infrastructure domain.

The certification of the Oracle FMW Infrastructure on Docker does not require the use of any file presented in this repository. Customers and users are welcome to use them as starters, and customize, tweak, or create from scratch, new scripts and Dockerfiles.
"

https://github.com/oracle/docker-images/tree/main/OracleFMWInfrastructure/dockerfiles/12.2.1.4

Adhering to existing standard. Please be informed that README.md has gone through our Documentation team.

File: OracleVeridata/12.2.1.4/README.md (hunk 48 above)
[the remark this answers is not in the document]
Comment (Djelibeybi) [reply OracleVeridata/12.2.1.4/README.md:37]: What instruction? There is no instruction.

[the remark this answers is not in the document]
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/README.md:37]: The latest file looks like this 
"
3. Oracle GoldenGate Veridata Installer:

Download the Oracle GoldenGate Veridata 12.2.1.4.0 Installer in to your local directory from the following location:

https://www.oracle.com/middleware/technologies/goldengate-downloads.html

"

File: OracleVeridata/12.2.1.4/LICENSE.txt (hunk 44 above)
[the remark this answers is not in the document]
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/LICENSE.txt:1]: Removed

File: OracleVeridata/12.2.1.4/README.md [hunk 51]
```diff
@@ -1,51 +1,58 @@
...
 
+=============================================================================
 
+## 2†Prerequisites
 
-## How to build
+Create Docker network to run Admin, Managed server, and repository Database (in case needed).
```
[the remark this answers is not in the document]
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/README.md]: Fixed

File: OracleVeridata/12.2.1.4/README.md [hunk 52]
```diff
@@ -1,51 +1,58 @@
...
 
-2. Oracle Database
+2. Oracle Database:
 
-Oracle Database is required to install OGG Veridata repository.
+Oracle Database is required to install Oracle GoldenGateVeridata repository.
 You can use an existing Oracle Database or build a Oracle Database image.
 
-For building a Oracle Database image use the following command
+To build an Oracle Database image run the following command:
```
[the remark this answers is not in the document]
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/README.md]: Fixed

File: OracleVeridata/12.2.1.4/README.md (hunk 47 above)
[the remark this answers is not in the document]
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/README.md]: Fixed

File: OracleVeridata/12.2.1.4/run_admin_server.sh (hunk 46 above)
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/run_admin_server.sh]: Merge issues fixed

File: OracleVeridata/12.2.1.4/run_admin_server.sh (hunk 45 above)
Comment (arnabnandioracle) [reply OracleVeridata/12.2.1.4/run_admin_server.sh]: Merge issue fixed

**Review by Djelibeybi — APPROVED** [review]

**Review by Djelibeybi — APPROVED** [review]

**Pull request merged**